satchless/payment/tests.py

```diff
@@ -5,84 +5,118 @@
-                                                            amount=0,
-                                                            name='test')
-        return payment_variant
+        return models.PaymentVariant.objects.create(
+            order=order,
+            name=typ,
+            price=0,
+            amount=0,
+        )
 
     def confirm(self, order, typ=None, variant=None):
-        pass
+        order.set_status('confirmed')
 
-class CreditCardNumberFieldTest(TestCase):
-    def setUp(self):
-        self.field = fields.CreditCardNumberField()
-
-    def test_valid_number(self):
-        TEST_NUMBERS = [
-            # VISA
-            '4111 1111 1111 1111',
-            # MasterCard
-            '5500 0000 0000 0004',
-            '5424 0000 0000 0015',
-            # American Express
-            '3782 8224 6310 005',
-            '3400 0000 0000 009',
-            '3700 0000 0000 002',
-            # Discover
-            '6011 1111 1111 1117',
-            '6011 0000 0000 0004',
-            '6011 0000 0000 0012',
-            # Diner's Club
-            '3000 0000 0000 04',
-            '2014 0000 0000 009',
-            # JCB
-            '3088 0000 0000 0009',
-        ]
-        for number in TEST_NUMBERS:
-            self.assertTrue(self.field.clean(number))
-
-    def test_invalid_number(self):
-        INVALID_NUMBERS = [
-            '9999 9999 9999 9999',
-            'abc',
-            '--',
-        ]
-        for number in INVALID_NUMBERS:
-            self.assertRaises(forms.ValidationError, self.field.clean,
-                              number)
-
-
-class CreditCardExpirationFieldTest(TestCase):
-    def setUp(self):
-        self.field = fields.CreditCardExpirationField()
 
-    def test_valid_date(self):
-        expires = datetime.date.today() + datetime.timedelta(days=30)
-        self.assertTrue(self.field.clean(expires))
+class TestPaymentProvider2Form(forms.Form):
+    amount = forms.IntegerField()
+
+    def __init__(self, **kwargs):
+        self.order = kwargs.pop('order')
+        self.typ = kwargs.pop('typ')
+        super(TestPaymentProvider2Form, self).__init__(**kwargs)
 
-    def test_past_date(self):
-        expires = datetime.date.today() + datetime.timedelta(days= -60)
-        self.assertRaises(forms.ValidationError, self.field.clean,
-                          expires)
+
+class TestPaymentProvider2(TestPaymentProvider):
+    def enum_types(self, order=None, customer=None):
+        yield self, TestPaymentType('platinum', 'Platinum', order=order, customer=customer)
+        yield self, TestPaymentType('gold-pressed-latinum', 'Gold-pressed latinum', order=order, customer=customer)
+
+    def get_configuration_form(self, order, data, typ=None):
+        return TestPaymentProvider2Form(data=data, order=order, typ=typ)
+
+    def create_variant(self, order, form, typ=None):
+        if not form.is_valid():
+            raise PaymentFailure(repr(form.errors))
```

Comment:
`PaymentFailure` is not defined, `form` must always be valid or the method not called in tests.